modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,334 @@
+
+    /**
+     * The image defaults to the official RabbitmQ image: <a href="https://hub.docker.com/_/rabbitmq/">RabbitMQ</a>.
+     */
+    private static final String DEFAULT_IMAGE_NAME = "rabbitmq";
+    private static final String DEFAULT_TAG = "3.7-management-alpine";
+
+    private static final int DEFAULT_AMQP_PORT = 5672;
+    private static final int DEFAULT_AMQPS_PORT = 5671;
+    private static final int DEFAULT_HTTPS_PORT = 15671;
+    private static final int DEFAULT_HTTP_PORT = 15672;
+
+    private String adminPassword = "guest";
+    private String adminUsername = "guest";
+    private final List<List<String>> values = new ArrayList<>();
+
+    /**
+     * Creates a Testcontainer using the official Neo4j docker image.
+     */
+    public RabbitMQContainer() {
+        this(DEFAULT_TAG);
+    }
+
+    /**
+     * Creates a Testcontainer using a specific docker image.
+     *
+     * @param tag The docker image to use.
+     */
+    public RabbitMQContainer(String tag) {
+        super(DEFAULT_IMAGE_NAME + ":" + tag);
+
+        this.waitStrategy = Wait.
+                forLogMessage(".*Server startup complete.*", 1)
+                .withStartupTimeout(Duration.ofSeconds(60));
+    }
+
+    @Override
+    public Set<Integer> getLivenessCheckPortNumbers() {
+
+        return Stream.of(DEFAULT_AMQP_PORT, DEFAULT_AMQPS_PORT, DEFAULT_HTTP_PORT, DEFAULT_HTTPS_PORT)
+                .map(this::getMappedPort)
+                .collect(toSet());
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPorts(DEFAULT_AMQP_PORT, DEFAULT_AMQPS_PORT, DEFAULT_HTTP_PORT, DEFAULT_HTTPS_PORT);
+
+        if (adminPassword != null) {
+            addEnv("RABBITMQ_DEFAULT_PASS", adminPassword);
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+
+        values.forEach(command -> {
+            try {
+                ExecResult execResult = execInContainer(command.toArray(new String[0]));
+                if (execResult.getExitCode() != 0) {
+                    logger().error("Could not execute command {}: {}", command, execResult.getStderr());
+                }
+            } catch (IOException | InterruptedException e) {
+                logger().error("Could not execute command {}: {}", command, e.getMessage());
+            }
+        });
+    }
+
+    public Integer getAmqpPort() {
+        return getMappedPort(DEFAULT_AMQP_PORT);
+    }
+
+    public Integer getAmqpsPort() {
+        return getMappedPort(DEFAULT_AMQPS_PORT);
+    }
+
+    public Integer getHttpsPort() {
+        return getMappedPort(DEFAULT_HTTPS_PORT);
+    }
+
+    public Integer getHttpPort() {
+        return getMappedPort(DEFAULT_HTTP_PORT);
+    }
+
+    /**
+     * @return AMQP URL for use with AMQP clients.
+     */
+    public String getAmqpUrl() {
+        return String.format("amqp://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_AMQP_PORT));
```

Comment:
I would suggest calling `getAmqpPort()` here (and in other `get*Url` methods) to avoid a duplication of this code

Reply:
Good idea. Done